### PR TITLE
correcting ticket's details view for admin.

### PR DESCRIPTION
**Pull request by rishabhs95**
fixing bug #877 .

### CODE REVIEW

File: app/views/admin/tickets/index.html.haml [hunk 1]
```diff
@@ -10,9 +10,9 @@
       %table.table.table-hover#tickets
         %thead
           %th Title
-          %th Description
           %th Price
           %th Sold
+          %th Total Price
```
Comment (hennevogel) [inline app/views/admin/tickets/index.html.haml]: It's not the total price. It's the **Turnover**

File: app/views/admin/tickets/index.html.haml [hunk 2]
```diff
@@ -21,11 +21,11 @@
...
                   = ticket.title
               %td
-                = truncate(ticket.description)
-              %td
                 = humanized_money_with_symbol ticket.price
               %td
-                = ticket.buyers.count
+                = ticket.ticket_purchases.inject(0){|sum,tp| sum+tp.quantity}
+              %td
+                = humanized_money_with_symbol ticket.ticket_purchases.inject(0){|sum,tp| sum+tp.quantity}*ticket.price
```
Comment (hennevogel) [inline app/views/admin/tickets/index.html.haml]: DRY this and move it to two methods (`#sold` and `#turnover`) in `Ticket`

File: app/views/admin/tickets/show.html.haml [hunk 3]
```diff
@@ -6,6 +6,11 @@
         Ticket
         %small
           = humanized_money_with_symbol @ticket.price
+      - if @ticket.description?
```
Comment (hennevogel) [inline app/views/admin/tickets/show.html.haml]: This is potentially very large. Make this a [tooltip or popover](http://getbootstrap.com/javascript/#tooltips) on `@ticket.title`

File: app/controllers/admin/tickets_controller.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@ module Admin
   class TicketsController < Admin::BaseController
     load_and_authorize_resource :conference, find_by: :short_title
     load_and_authorize_resource :ticket, through: :conference
+    helper_method :tickets_sold, :tickets_turnover
```
Comment (ChrisBr) [inline app/controllers/admin/tickets_controller.rb]: Huh? Why? Why not a model method?

File: app/assets/javascripts/application.js [hunk 5]
```diff
@@ -49,4 +49,7 @@ $(document).ready(function() {
     $('body').smoothScroll({
         delegateSelector: 'a.smoothscroll'
     });
+    $(function () {
```
Comment (ChrisBr) [inline app/assets/javascripts/application.js]: Please don't to this in the application.js 
``
For performance reasons, the Tooltip and Popover data-apis are opt-in, meaning you must initialize them yourself.
``

Comment (rishabhs95) [inline app/assets/javascripts/application.js]: should i shift it to `osem-tickets.js` then?

File: app/controllers/admin/tickets_controller.rb (hunk 4 above)
Comment (rishabhs95) [inline app/controllers/admin/tickets_controller.rb]: yes... i just noticed it is better to shift most of the logic to models :+1:

File: app/models/ticket.rb [hunk 6]
```diff
@@ -48,6 +48,13 @@ def self.total_price(conference, user)
     result ? result : Money.new(0, 'USD')
   end
 
+  def self.tickets_turnover(ticket_id, conference_id)
```
Comment (ChrisBr) [inline app/models/ticket.rb]: Ever heard from instance methods ;)
http://www.railstips.org/blog/archives/2009/05/11/class-and-instance-methods-in-ruby/

File: app/models/ticket_purchase.rb [hunk 7]
```diff
@@ -53,4 +53,10 @@ def self.update_quantity(conference, quantity, ticket, user)
     purchase.quantity = quantity if quantity > 0
     purchase
   end
+
+  def self.tickets_sold(ticket_id, conference_id)
```
Comment (ChrisBr) [inline app/models/ticket_purchase.rb]: This method also belongs to ticket and probably you can do the inject in a database query for better performance ...

File: app/models/ticket.rb [hunk 8]
```diff
@@ -48,6 +48,15 @@ def self.total_price(conference, user)
     result ? result : Money.new(0, 'USD')
   end
 
+  def tickets_sold
+    TicketPurchase.where(ticket_id: id, conference_id: conference_id).inject(0) { |a, e| a + e.quantity }
```
Comment (ChrisBr) [inline app/models/ticket.rb]: ``ticket_purchases.sum(:quantity)`` should also work!
(Note it's untested...)

File: app/models/ticket.rb [hunk 9]
```diff
@@ -48,6 +48,15 @@ def self.total_price(conference, user)
...
+  end
+
+  def tickets_turnover
+    price = Money.new(price_cents, price_currency)
```
Comment (ChrisBr) [inline app/models/ticket.rb]: I assume that you don't need this line, just ``tickets_sold * price`` should be fine!

**Comment by rishabhs95** [comment]
pushed with all suggested changes

**Comment by ChrisBr** [comment]
LGTM

**Pull request merged**